src/FrameBuffer.cpp

```diff
@@ -538,6 +538,11 @@ void FrameBufferList::saveBuffer(u32 _address, u16 _format, u16 _size, u16 _widt
 	if (m_pCurrent != NULL) {
 		// Correct buffer's end address
 		if (!m_pCurrent->isAuxiliary()) {
+			if(m_prevColorImageHeight == 0)
+			{
+				m_prevColorImageHeight = VI.height;
+			}
+			
 			if (gDP.colorImage.height > 200)
```

Comment:
This is buffer height correction code. Initially buffer's height is VI.height for main buffer. It is good assumption.
If we calculated buffer height more precisely during rendering and gDP.colorImage.height is not zero, we can correct actual end address of the buffer. It is very important for fb emulation to know actual address space of the buffer.
If gDP.colorImage.height is zero, m_endAddress will not be corrected. It is ok.